PID logic sync to betaflight

borisbstyle opened the pull request:
Sync PID logic to latest betaflight.

Quick Summary:
- Replace median filter by moving average. 8 point is good for very low looptimes. Median can cause some noise under certain frequencies
- Removed extra accumulation from Iterm, which was experimental.
- Delta calculation for Yaw is now derived from gyro instead from rateError. This should decrease noise on yaw axis, which tends to be more noisy. D can now be used more safely on Yaw axis

borisbstyle committed:
PID logic sync to betaflight

borisbstyle commented:
Going to close this one. It is outdated and not what we need

rs2k commented:
I just got back from Vegas last night. I need to catch up on things again. We tested this a few days ago and found it did not improve flight characteristics.

borisbstyle commented:
No it is not to improve.

Median filter can cause artifacts in D where moving average did better job.
That means less prone to oscillations.

But anyway both of these cant beat the new filter i added now to
betaflight...bb logs are butter smooth. Thats why i removed
Op 13 jan. 2016 18:05 schreef "rs2k" notifications@github.com:

> I just got back from Vegas last night. I need to catch up on things again.
> We tested this a few days ago and found it did not improve flight
> characteristics.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/rs2k/raceflight/pull/39#issuecomment-171365222.
